### DISCUSSION TITLE
How do I troubleshoot "Cycle detected" issue?

### DISCUSSION THREAD

**Discussion by niyue** [microsoft/vcpkg#20485] [General]
I tried to use custom vcpkg registry to host several private packages, and use them in vcpkg manifest in my project. Most of my custom dependencies can be used easily, but I find one of the dependency cannot be installed, and vcpkg reported error like below:
```
vcpkg install
Error: Cycle detected during arrow:x64-osx:
my-custom-package:x64-osx
```

In `my-custom-package`'s `vcpkg.json` file, I specify `arrow` as one of its dependency, and I use my custom version of `arrow` instead of vcpkg's built-in arrow package, and I host the custom version of `arrow` in my custom vcpkg registry as well, but apparently, it doesn't depend on `my-custom-package`. I have no idea where this circular dependency comes from. I tried using `vcpkg depend-info` but I cannot see anything helpful to address this issue. 

Could some one shed some light on how I can troubleshoot such an issue?  Currently, the information given by vcpkg is not enough for me to figure out what it is complaining.

**Comment by mathisloge** [comment]
Have you added the custom registry and the package `arrow` in your `vcpkg-configuration.json`?
If yes, does your custom `arrow` uses a different versioning scheme then the offical `arrow`?

**Comment by niyue** [comment]
Yes, I added `arrow` to my custom registry. 

> does your custom arrow uses a different versioning scheme then the offical arrow

I name them differently (the official one is 5.0.0, and the custom one I added is 6.0.0.20210925), I am not sure if this qualifies as different versioning scheme?

I came up with a minimum project reproducing this issue and reported a bug here:
https://github.com/microsoft/vcpkg/issues/20503
